### PR TITLE
feat(backup/s3): add required configuration

### LINKED ISSUE #10220: Backup store can be configured
Allow configuring:
* bucket name
* AWS client credentials

### PR DESCRIPTION
**Pull request by lenaschoenburg**
* Adds a config value for client credentials.
* Adds a config value for the S3 endpoint.
* Adds a config value for bucket region.
* The store implementation can now build its own client, based on the given config.

closes #10220

### CODE REVIEW

File: backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java [hunk 1]
```diff
@@ -343,7 +355,18 @@ private CompletableFuture<PutObjectResponse> saveNamedFile(
         AsyncRequestBody.fromFile(filePath));
   }
 
-  public static String objectPrefix(BackupIdentifier id) {
-    return "%s/%s/%s/".formatted(id.partitionId(), id.checkpointId(), id.nodeId());
+  private static S3AsyncClient buildClient(S3BackupConfig config) {
```
Comment (deepthidevaki) [inline backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java:358]:
```suggestion
  private static S3AsyncClient buildClient(final S3BackupConfig config) {
```

File: backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java [hunk 2]
```diff
@@ -72,11 +76,19 @@ public final class S3BackupStore implements BackupStore {
...
+  public S3BackupStore(final S3BackupConfig config) {
+    this(config, buildClient(config));
+  }
+
   public S3BackupStore(final S3BackupConfig config, final S3AsyncClient client) {
     this.config = config;
     this.client = client;
   }
 
+  public static String objectPrefix(BackupIdentifier id) {
```
Comment (deepthidevaki) [inline backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java:88]:
```suggestion
  public static String objectPrefix(final BackupIdentifier id) {
```

File: backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupConfig.java [hunk 3]
```diff
@@ -7,5 +7,51 @@
...
+    Optional<Credentials> credentials) {
+
+  /**
+   * Creates a config without setting the region and credentials.
+   *
+   * @param bucketName Name of the backup that will be used for storing backups
+   * @see S3BackupConfig#S3BackupConfig(String bucketName, Optional endpoint, Optional region,
+   *     Optional credentials)
+   */
+  public S3BackupConfig(String bucketName) {
```
Comment (deepthidevaki) [inline backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupConfig.java:40]:
```suggestion
  public S3BackupConfig(final String bucketName) {
```

File: backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java [hunk 4]
```diff
@@ -343,7 +355,18 @@ private CompletableFuture<PutObjectResponse> saveNamedFile(
...
+    config.region().ifPresent(region -> builder.region(Region.of(region)));
+    config
+        .credentials()
+        .ifPresent(
+            credentials ->
+                builder.credentialsProvider(
+                    StaticCredentialsProvider.create(
+                        AwsBasicCredentials.create(
+                            credentials.accessKey(), credentials.secretKey()))));
+    return builder.build();
```
Comment (deepthidevaki) [inline backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java:370]: :question: What would happen if there are no credential provided and no credentials available from environment or other ways. Does the build fails? or does it fail when the first time it tries to access S3?

**Review by deepthidevaki — APPROVED** [review]
:rocket:

File: backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java (hunk 1 above)
Comment (oleschoenburg) [reply backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java:358]: Huh, there are a bunch of missing finals. I thought these were added automatically by IntelliJ :thinking:

**Pull request merged**